### DISCUSSION TITLE
I can't generate the ssl certificate

### DISCUSSION THREAD

**Discussion by arthur32810** [OvenMediaLabs/OvenMediaEngine#1437] [Help] [answered]
Hello,

We are setting up an OvenMediaEngine server, and everything works perfectly with HTTP.

Our problem is generating and installing the SSL certificate, we don't know how to do it. We did look at the documentation that explains how to add the certificate, but it doesn't point to a link explaining how to create it. Do you have a resource to share with me that could help us generate a certificate and configure it?

I thank you in advance,
Best regards,
Arthur

**Comment by naanlizard** [comment, marked as answer]
Google letsencrypt. Alternatively you might be able to buy one from your dns provider or domain registrar. Lastly you could generate a self signed certificate but that's really only useful for testing and kind of a pain in the butt

**Reply by arthur32810** [reply]
Hello @naanlizard , thank you for your reply. Are you using lets encrypt to generate the certificate and add it to the OME server in the TLS part?

**Reply by naanlizard** [reply]
Correct, letsencrypt is fantastic and has plenty of documentation if you Google

**Reply by arthur32810** [reply]
Is it installed on the workstation or on the Internet box?

**Reply by naanlizard** [reply]
I'm not trying to be rude, but I think either we've got a translation error or you need to do more reading on typical server setups. 

What does your infrastructure look like right now?

**Reply by naanlizard** [reply]
Maybe more importantly, what is your end goal

**Reply by arthur32810** [reply]
Thanks for your comments, I'll try to be more precise:

The OvenMediaEngine server runs on ubuntu 22.04.
The aim is to output to the Internet in wss to display the feed on our website.

**Reply by naanlizard** [reply]
Ok, generally speaking you want to have letsencrypt run on the server that will use the certificate. It's just easiest that way. You should Google something like "how to generate an SSL certificate with letsencrypt" and read + understand a guide, then follow it. You'll probably want to expose a few ports from your server to be able to generate the certificate, and you'll need a domain to point to the IP of the server of course to generate the certificate for

**Reply by arthur32810** [reply]
thank you, I'll follow what you told me, I'll keep you informed of my progress.

**Reply by arthur32810** [reply]
Hello,

I finally managed to place the certificate!
Thanks @naanlizard  for your help
